### PR TITLE
Proof of concept in Python, on behalf of SDA SE

### PR DESCRIPTION
**Pull request by hans4dev**
TODO:

- [x]  fix threat-model to file distribution (for demo only 1 model written to 1 file)
- [x]  read `.env` file for credentials instead of `CONFLUENCE_` environment variables
- [x]  simplify model ~~(optionally make attribute configurable)~~
- [x]  add header to YAML denoting the auto-generation 
- [ ]  parameterize with `Click` or `Typer` (output-path, application-map, search label etc.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Meta information about a threat modeling activity
+
+<quote>
+Information identifying the threat model typically includes the following:
+1. Application Name: The name of the application examined.
+2. Application Version: The version of the application examined.
```
Comment (wurstbrot) [inline README.md:6]: Can the version be optional?

File: confluence_collector.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+def write_yaml_files(folder, subject_threatmodel_tuples):
+    s = space_to_application_map['MR']  # FIXME: remove the restriction
+    files_written = 0
+
+    # TODO: loop through each element of the tuples, group files by application
+    model = ConductionOfSimpleThreatModelingOnTechnicalLevel(
+        components=[t[1] for t in subject_threatmodel_tuples if t[0] == s])  # FIXME: remove the restriction
+    os.makedirs(f"{folder}/{s.team_name}", exist_ok=True)
+    output_filename = f"{folder}/{s.team_name}/{s.application_name}_application.yaml"
```
Comment (wurstbrot) [inline confluence_collector.py:95]: maybe this file shoud have the pre- or postfix `automated` or `collector`.

File: docs/yaml.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+But, the model generation resulted in a known issue: [Infinite recursion when "$ref" at root. · Issue #92 · jacksmith15/statham-schema](https://github.com/jacksmith15/statham-schema/issues/92).
+
+## YAML output, parsing and serialization
+For basic and low-level YAML operations we could use [ruamel.yaml](https://pypi.org/project/ruamel.yaml/) (MIT) is a YAML 1.2 loader/dumper package for Python.
+But when we want to dump a Pydantic model that was generated beforehand, then we should use [NowanIlfideme/pydantic-yaml](https://github.com/NowanIlfideme/pydantic-yaml): YAML support for Pydantic models.
+
+Example output:
+```yaml
+# in folder <team-name>, filename <automated_threat-modeling>.yaml
```
Comment (wurstbrot) [inline docs/yaml.md:29]: A header like 
```
# this file is automatically created. Manual entries are normally not needed.
```
would be nice.

File: model.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+class Link(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    title: str
+    url: AnyUrl
+
+
+class ReductionOfTheAttackSurfaceComponent(BaseModel):
```
Comment (wurstbrot) [inline model.py:21]: I would not put the hole model in the collector. Only the one needed (in this case threat modeling)

File: model.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+class Settings(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    desired_level: str = Field(..., alias='desired level')
+    application: str
+    team: str
+
+
+class ConductionOfSimpleThreatModelingOnTechnicalLevelComponent(BaseModel):
```
Comment (wurstbrot) [inline model.py:45]: In addition to `ConductionOfSimpleThreatModelingOnTechnicalLevelComponent` DSOMM has `Conduction of simple threat modeling on business level` or `Conduction of advanced threat modeling`.

The level often depends on 
- quality
- frequency

I wouldn't focus in the start on quality and just on frequency. The frequency/level shouldn't influence the name which leads to a generic name like `conduction of threat modeling`.

File: README.md (hunk 1 above)
Comment (hans4dev) [reply README.md]: Yes, of course. 
This was just a related block-quote from an OWASP article. 

Improved the README and stated our information requirements.

File: confluence_collector.py (hunk 2 above)
Comment (hans4dev) [reply confluence_collector.py]: Sure, we should denote the "autogenerated" nature and source. 
Thanks to your suggestion of a file-header, this is now done within the file.
So we can keep the file-names simple and short.

File: docs/yaml.md (hunk 3 above)
Comment (hans4dev) [reply docs/yaml.md:29]: Great idea, added the comment to files.

File: model.py (hunk 4 above)
Comment (hans4dev) [reply model.py]: simplified the model.

**Review by wurstbrot — APPROVED** [review]

**Pull request merged**